chaoslib/litmus/platform/gke/node_memory_consumption.yml

```diff
@@ -0,0 +1,121 @@
+ ## GETTING THE PRE CHAOS NODE NAME OF APPLICATION 
+- block:
+  
+    - name: Getting the pre-chaos application pod name 
+      k8s_facts:
+        kind: Pod
+        label_selectors:
+          - "{{ a_label }}"
+      register: app_pod_name
+    
+    ## RECORDING THE NODE NAME ON WHICH APPLICATION IS RUNNING
+    - name: Recording the node name of application pod
+      set_fact:
+        node_name: "{{ app_pod_name | json_query('resources[0].spec.nodeName')}}"
+
+    ## RECORD EVENT FOR CHAOS INJECTION
+    - include_tasks: /utils/common/generate-kubernetes-chaos-events.yml
+      vars:
+        stage: "ChaosInject"
+        exp_pod_name: "{{ chaos_pod_name }}"
+        engine_ns: "{{ a_ns }}"
+        message: "Injecting {{ c_experiment }} chaos on {{ node_name }} node"
+      when: "c_engine is defined and c_engine != ''"
+        
+    ## PATCH THE ENGINE UID
+    - name: Patch the engine uid
+      template:
+        src:  /chaoslib/litmus/platform/gke/node-memory-hog-job.j2
+        dest: /chaoslib/litmus/platform/gke/node-memory-hog-job.yml
+      vars:
+        stress_time: "{{ c_duration }}s"
+        total_memory_percentage: "{{ memory_percentage }}%"
+
+    ## WAIT FOR THE SPECIFIED RAMP TIME
+    - name: Wait for the specified ramp time before injecting chaos
+      wait_for: timeout="{{ ramp_time }}"
+      when: "ramp_time is defined and ramp_time != ''"
+
+    ## CREATING JOB FOR MEMORY-HOG
+    - name: Creating Job for memory-hog
+      k8s:
+        state: present
+        namespace: "{{ a_ns }}"
+        src: /chaoslib/litmus/platform/gke/node-memory-hog-job.yml
+      register: memory_hog_job_result
+
+    ## CHECKING THE STATUS OF JOB
+    - name: Checking the status of Job
```

Comment:
After creation of job can you add a task which will wait till the completion of job.  Refer https://github.com/litmuschaos/litmus/blob/master/chaoslib/pumba/network_chaos/network_chaos.yml#L89
After that no need for this task